server/data/data-postgres/src/main/java/org/apache/james/domainlist/postgres/PostgresDomainList.java

```diff
@@ -0,0 +1,86 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.domainlist.postgres;
+
+import static org.apache.james.domainlist.postgres.PostgresDomainModule.PostgresDomainTable.DOMAIN;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.apache.james.backends.postgres.utils.PostgresExecutor;
+import org.apache.james.core.Domain;
+import org.apache.james.dnsservice.api.DNSService;
+import org.apache.james.domainlist.api.DomainListException;
+import org.apache.james.domainlist.lib.AbstractDomainList;
+import org.jooq.exception.DataAccessException;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class PostgresDomainList extends AbstractDomainList {
+    private final PostgresExecutor postgresExecutor;
+
+    @Inject
+    public PostgresDomainList(DNSService dnsService, PostgresExecutor postgresExecutor) {
+        super(dnsService);
+        this.postgresExecutor = postgresExecutor;
+    }
+
+    @Override
+    public void addDomain(Domain domain) throws DomainListException {
+        try {
+            postgresExecutor.executeVoid(dslContext -> Mono.from(dslContext.insertInto(PostgresDomainModule.PostgresDomainTable.TABLE_NAME, DOMAIN)
+                    .values(domain.asString())))
+                .block();
+        } catch (DataAccessException exception) {
+            throw new DomainListException(domain.name() + " already exists.");
+        }
+    }
+
+    @Override
+    protected List<Domain> getDomainListInternal() throws DomainListException {
+        return postgresExecutor.executeRows(dsl -> Flux.from(dsl.selectFrom(PostgresDomainModule.PostgresDomainTable.TABLE_NAME)))
+            .map(record -> Domain.of(record.get(DOMAIN)))
+            .collectList()
+            .block();
+    }
+
+    @Override
+    protected boolean containsDomainInternal(Domain domain) throws DomainListException {
+        return postgresExecutor.executeRow(dsl -> Mono.from(dsl.selectFrom(PostgresDomainModule.PostgresDomainTable.TABLE_NAME)
+            .where(DOMAIN.eq(domain.asString()))))
+            .blockOptional()
+            .isPresent();
+    }
+
+    @Override
+    protected void doRemoveDomain(Domain domain) throws DomainListException {
+        boolean executed = postgresExecutor.executeRow(dslContext -> Mono.from(dslContext.deleteFrom(PostgresDomainModule.PostgresDomainTable.TABLE_NAME)
+                .where(DOMAIN.eq(domain.asString()))
+                .returning(DOMAIN)))
+            .blockOptional()
+            .isPresent();
+
+        if (!executed) {
+            throw new DomainListException(domain.name() + " already exists.");
```

Comment:
```suggestion
            throw new DomainListException(domain.name() + " was not found");
```

Reply:
Good catch. Was initially good but I battled a while and mixed up in the end. Will fix :)